python/az/aro/azext_aro/custom.py

```diff
@@ -138,43 +150,51 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
-        worker_subnets = {w.subnet_id for w in oc.worker_profiles}
-
-        master_parts = parse_resource_id(master_subnet)
-        vnet = resource_id(
-            subscription=master_parts['subscription'],
-            resource_group=master_parts['resource_group'],
-            namespace='Microsoft.Network',
-            type='virtualNetworks',
-            name=master_parts['name'],
-        )
+        # Get associated cluster resources we need to assign network contributor on
+        resources = get_cluster_network_resources(cmd.cli_ctx, oc)
 
         aad = AADManager(cmd.cli_ctx)
         rp_client_id = os.environ.get('AZURE_FP_CLIENT_ID', FP_CLIENT_ID)
-        rp_client_sp = aad.get_service_principal(rp_client_id)
+        rp_client_sp = None
 
+        # Best effort call - assume the role assignments on the SP exist if exception raised
+        try:
+            rp_client_sp = aad.get_service_principal(rp_client_id)
+        except GraphErrorException:
+            pass
+
+    except (HttpOperationError, CloudError) as e:
+        logger.info(e.message)
+
+    try:
         # Customers frequently remove the RP's permissions, then cannot
         # delete the cluster. Where possible, fix this before attempting
         # deletion.
         if rp_client_sp:
-            sp_id = rp_client_sp.object_id
-            assign_contributor_to_vnet(cmd.cli_ctx, vnet, sp_id)
-            assign_contributor_to_routetable(cmd.cli_ctx,
-                                             worker_subnets | {master_subnet},
-                                             sp_id)
-    except (CloudError, HttpOperationError) as e:
-        # Default to old deletion behaviour in case operations throw an
-        # exception above. Log the error.
-        logger.info(e.message)
+            for resource in sorted(resources):
+                # Create role assignments if they don't exist
+                # Note: This logic assumes that if we cannot if the role assignment
+                # we assume that it already exists
+                resource_contributor_exists = True
 
-    return sdk_no_wait(no_wait, client.delete,
-                       resource_group_name=resource_group_name,
-                       resource_name=resource_name)
+                try:
+                    resource_contributor_exists = has_network_contributor_on_resource(cmd.cli_ctx, resource,
+                                                                                      rp_client_sp.object_id)
+                except CloudError:
+                    continue
+
+                if not resource_contributor_exists:
+                    assign_network_contributor_to_resource(cmd.cli_ctx, resource, rp_client_sp.object_id)
+
+        return sdk_no_wait(no_wait, client.delete,
+                           resource_group_name=resource_group_name,
+                           resource_name=resource_name)
+
+    except (HttpOperationError, CloudError) as e:
```

Comment:
why are we wrapping this portion of the code in try/except?
1) sdk_no_wait should not be wrapped by try/except
2) has_network_contributor_on_resource is correctly wrapped by try/except already - what other error conditions are we worrying about here?